garden_ai/utils.py

```diff
@@ -176,3 +179,38 @@ def extract_email_from_globus_jwt(jwt: str) -> str:
     except KeyError as e:
         raise Exception("JWT did not include user email") from e
     return email
+
+
+def read_conda_deps(conda_file: str) -> Tuple[Optional[str], List[str], List[str]]:
+    """parse the dependencies from the given file and return as a (python_version, conda_dependencies, pip_dependencies) tuple
+    python_version defaults to the one described by sys.version_info
+
+    """
+
+    with open(conda_file, "r") as f:
+        contents = yaml.safe_load(f.read())
+
+    python_spec_re = re.compile(r"python[=<>! ]*")
+    pip_spec_re = re.compile(r"pip[=<>!]?")
+
+    # python_version = ".".join(map(str, sys.version_info[:3]))
+    python_version = None
+    conda_dependencies = []
+    pip_dependencies = []
+
+    for dependency in contents["dependencies"]:
```

Comment:
Are being too trusting of .yml files we're being fed? If this yml file is not a compliant conda yaml, can we wrap this in some exception so that we're not just passing up a KeyError to the user on `contents["dependencies"]`?